move/sources/booster.move

```diff
@@ -0,0 +1,148 @@
+module refund::booster {
+    // use std::debug::print;
+    use sui::tx_context::{TxContext, sender};
+	use sui::coin::{Self, Coin};
+    use sui::transfer;
+    use sui::balance;
+    use sui::table;
+    use sui::dynamic_field as df;
+    use sui::object::{Self, UID};
+    use sui::package::Publisher;
+    use sui::sui::SUI;
+    
+    use refund::refund::{
+        Self, RefundPool, claim_refund_,
+        accounting, unclaimed, uid_mut,
+        accounting_mut, booster_pool_mut
+    };
+    use refund::accounting::{
+        total_raised_for_boost, total_raised_for_boost_mut,
+        current_liabilities, total_boosted_mut
+    };
+    use refund::math::div;
+    use refund::table::{Self as refund_table};
+    use refund::pool::{funders_mut, funds_mut};
+
+    const EAddressRetrievedBoostCap: u64 = 0;
+
+    struct BoostedCapDfKey has copy, store, drop { affected_address: address }
+
+    struct BoostedClaimCap has key {
+        id: UID,
+        new_address: address
+    }
+
+    // === Phase 2: Funding ===
+
+    public entry fun fund(
+        pool: &mut RefundPool,
+        coin: Coin<SUI>,
+        ctx: &mut TxContext,
+    ) {
+        refund::assert_funding_phase(pool);
+        
+        let amount = coin::value(&coin);
+        let total_raised_for_boost = total_raised_for_boost_mut(accounting_mut(pool));
+        *total_raised_for_boost = *total_raised_for_boost + amount;
+
+        let booster = booster_pool_mut(pool);
+        refund_table::insert_or_add(funders_mut(booster), sender(ctx), amount);
+        balance::join(funds_mut(booster), coin::into_balance(coin)); 
+    }
+
+    public entry fun withdraw_funds(
+        pool: &mut RefundPool,
+        amount: u64,
+        ctx: &mut TxContext,
+    ) {
+        refund::assert_funding_phase(pool);
+
+        let total_raised_for_boost = total_raised_for_boost_mut(accounting_mut(pool));
+        *total_raised_for_boost = *total_raised_for_boost - amount;
+
+        let booster = booster_pool_mut(pool);
+        refund_table::remove_or_subtract(funders_mut(booster), sender(ctx), amount);
+        let funds = coin::from_balance(balance::split(funds_mut(booster), amount) , ctx);
+        transfer::public_transfer(funds, sender(ctx));
+    }
+
+    // === Phase 3: Claim Refund ===
+
+    // called via tx sent from Rinbot address
+    public fun allow_boosted_claim(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        affected_address: address,
+        new_address: address,
```

Comment:
We're creating a `BoosterClaimCap` that will transfer ALL the refund (not only the 50% booster) to the "new_address" passed by the publisher. 

This could be the required logic here but wanted to flag just in case (since the user has no explicit control on where funds end up on-chain).

Reply:
Here's the most updated implementation:

```
public entry fun claim_refund_boosted(
    cap: BoostedClaimCap,
    pool: &mut RefundPool,
    ctx: &mut TxContext,
) {
    let affected_address = sender(ctx);
    refund::assert_address(pool, affected_address);

    let BoostedClaimCap { id, new_address } = cap;
    object::delete(id);
    
    let refund_amount = *table::borrow(unclaimed(pool), affected_address);

    // Base Refund
    let refund = claim_refund_(pool, affected_address, ctx);

    // Booster Refund
    let boost = div(refund_amount, 2);

    let total_boosted = total_boosted_mut(accounting_mut(pool));
    *total_boosted = *total_boosted + boost;

    let booster = booster_pool_mut(pool);
    let boosted_funds = balance::split(funds_mut(booster), boost);

    balance::join(coin::balance_mut(&mut refund), boosted_funds);

    transfer::public_transfer(refund, new_address);
}
```


We're claiming the 100% refund here:

`let refund = claim_refund_(pool, affected_address, ctx);`

We then add the boost on top of it:

```
// Booster Refund
        let boost = div(refund_amount, 2);

        let total_boosted = total_boosted_mut(accounting_mut(pool));
        *total_boosted = *total_boosted + boost;

        let booster = booster_pool_mut(pool);
        let boosted_funds = balance::split(funds_mut(booster), boost);
```

Join both balances and send it to `new_address`